benchmarking/runs/priority_algorithm/internal_parameter_exploration/five_diversity_constraint.py

```diff
@@ -0,0 +1,228 @@
+        max_iterations_range = [1, 5, 10, 20, 30]
+
+        scenario = FiveDiversityConstraintScenario(
+            value_of_female=Gender.FEMALE.value,
+            value_of_a=Gpa.A.value,
+            value_of_21=21,
+            value_of_math=Major.MATH.value,
+            value_of_european=Race.European.value,
+        )
+        metrics = {
+            "PrioritySatisfaction": PrioritySatisfaction(
+                goals_to_priorities(scenario.goals),
+                False,
+            ),
+            "AverageCosineDifference": AverageCosineDifference(),
+            "AverageSoloStatus": AverageSoloStatus(
+                minority_groups={
+                    ScenarioAttribute.GENDER.value: [_.value for _ in Gender.values()],
+                    ScenarioAttribute.GPA.value: [_.value for _ in Gpa.values()],
+                    ScenarioAttribute.AGE.value: [_.value for _ in Age.values()],
+                    ScenarioAttribute.MAJOR.value: [_.value for _ in Major.values()],
+                    ScenarioAttribute.RACE.value: [_.value for _ in Race.values()],
+                }
+            ),
+        }
+        start_types = [
+            PriorityAlgorithmStartType.WEIGHT,
+            PriorityAlgorithmStartType.RANDOM,
+        ]
+
+        artifacts_dict = {
+            class_size: {start_type: {} for start_type in start_types}
+            for class_size in class_sizes
+        }
+        for class_size in class_sizes:
+            for start_type in start_types:
+                artifact: SimulationSetArtifact = SimulationSet(
+                    settings=SimulationSettings(
+                        num_teams=num_teams,
+                        scenario=scenario,
+                        student_provider=student_providers[class_size],
+                        cache_key=f"priority_algorithm/internal_parameter_exploration/class_size_{class_size}/five_diversity_constraint/25_buckets",
+                    ),
+                    algorithm_set={
+                        AlgorithmType.PRIORITY: [
+                            PriorityAlgorithmConfig(
+                                MAX_KEEP=max_keep,
+                                MAX_SPREAD=max_spread,
+                                MAX_ITERATE=max_iterations,
+                                MAX_TIME=1000001,
+                                START_TYPE=start_type,
+                                name=f"max_keep_{max_keep}-max_spread_{max_spread}-max_iterations_{max_iterations}-{start_type.value}_start",
+                            )
+                            for max_keep in max_keep_range
+                            for max_spread in max_spread_range
+                            for max_iterations in max_iterations_range
+                        ]
+                    },
+                ).run(num_runs=num_trials)
+
+                for name, simulation_artifact in artifact.items():
+                    match = re.search(
```

Comment:
nit for learning;

regex was probably not needed here lmao, could have just looped

Reply:
Yeah, I was definitely being greedy